### ISSUE TITLE
Use TagGroup ffrom react-aria-components for filters

### ISSUE THREAD

**Issue by radoslavzeman** [bratislava/marianum.sk#329]
Filter tags on search page should be implemented using new TagGroup components from `react-aria-components` to ensure accessibility.

![image](https://github.com/bratislava/marianum.sk/assets/19418224/f8ef286b-df7d-4d55-bb5e-521f327b0ae9)

Check bratislava.sk project, article filters.
